client/concordclient/include/client/concordclient/concord_client.hpp

```diff
@@ -130,14 +132,59 @@ class ConcordClient {
 
   // Register a callback that gets invoked for every validated event received.
   // void callback(EventGroup);
+  // Return subscriber ID used to unsubscribe.
   template <class CallbackT>
   void subscribe(const SubscribeRequest& request,
                  const std::unique_ptr<opentracing::Span>& parent_span,
                  CallbackT callback);
+
+  // Unsubscribe with the subscriber ID retrieved from a prior call to subscribe.
+  // Note, if the caller doesn't unsubscribe and no runtime error occurs then resources
+  // will be occupied forever.
   void unsubscribe();
 
  private:
+  logging::Logger logger_;
   ConcordClientConfig config_;
+
+  // TODO: Allow multiple subscriptions
+  std::atomic_bool stop_subscriber_{true};
+  std::thread subscriber_;
 };
 
+template <class CallbackT>
+void ConcordClient::subscribe(const SubscribeRequest& request,
+                              const std::unique_ptr<opentracing::Span>& parent_span,
+                              CallbackT callback) {
+  if (not stop_subscriber_) {
+    LOG_ERROR(logger_, "subscription already in progress - unsubscribe first");
+    return;
+  }
+
+  stop_subscriber_ = false;
+  std::atomic_bool* stop = &stop_subscriber_;
```

Comment:
The pointer is redundant here, a regular reference will be enough because accessing an atomic variable through a reference doesn't introduce any additional accesses or modifications, because references do not occupy memory locations. Therefore, performing potentially concurrent atomic operations is still safe when those operations occur through references.

Reply:
I think I misunderstood your comment to get rid of the extra variable. Nvm. That works. I will change it. Thanks!

Reply:
I think I misunderstood your comment to remove the extra variable. Nvm. That works. Thanks!